ui/src/kapacitor/components/RuleGraphDygraph.tsx

```diff
@@ -39,21 +39,32 @@ interface State {
 
 @ErrorHandling
 class RuleGraphDygraph extends Component<Props, State> {
-  public async componentWillReceiveProps(nextProps: Props) {
-    const {loading, timeSeries} = this.props
-    if (
-      loading !== nextProps.loading &&
-      nextProps.loading === RemoteDataState.Done
-    ) {
-      const result = await timeSeriesToDygraph(timeSeries, 'rule-builder')
-      this.setState({timeSeriesToDygraphResult: result})
+  constructor(props) {
+    super(props)
+
+    this.state = {}
```

Comment:
Would it be better to have initial state contain `timeSeriesToDygraphResult: null` instead of changing the shape of state?